BannerlordTwitch/BannerlordTwitch/Api/BLTApi.cs

```diff
@@ -0,0 +1,148 @@
+                )
+                {
+                    Trace.WriteLine("[EXEC VISITED]" + bltApiUrl.url);
+                    try { 
+                        bltApiUrl.onVisit.Invoke(req,resp);
+                    }catch(Exception e){
+                        Log.LogFeedSystem("Issue on visit api urls from BLTApiUrl : " + bltApiUrl.url);
+                    }
+                }
+            });
+        }
+
+        public void OnApiVisit(HttpListenerRequest req, HttpListenerResponse resp)
+        {
+            dynamic json = new ExpandoObject();
+            string url = HttpUtility.UrlDecode(req.Url.ToString());
+            json.url = url;
+            SendJSON(resp, json);
+        }
+        public void OnCommand(HttpListenerRequest req, HttpListenerResponse resp)
+        {
+            dynamic json = new ExpandoObject();
+
+            string cmdName = "";
+            string pseudo = "";
+            string args = "";
+
+            try
+            {
+                var splittedUrl = req.Url.ToString().Split('?');
+                //if url have option
+                var splittedOption = splittedUrl[1].Split('&');
+                //foreach option
+                for (int i = 0; i < splittedOption.Length; i++)
+                {
+                    string[] option = splittedOption[i].Split('=');
+                    //check key & value
+                    if (option[0] != null && option[1] != null)
+                    {
+                        switch (option[0])
+                        {
+                            case "cmd":
+                                cmdName = option[1];
+                                break;
+                            case "pseudo":
+                                pseudo = option[1];
+                                break;
+                            case "args":
+                                args = option[1];
+                                break;
+                            default:
+                                break;
+                        }
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                json.error = "Missing variables";
+                json.expected = "cmd=string&pseudo:string&args:string";
+                json.actualParams = "cmd=" + cmdName + "&pseudo=" + pseudo + "&args=" + args;
+            }
+
+            if (cmdName == "" || pseudo == "" || args == "")
+            {
+                json.error = "Missing variables";
+                json.expected = "cmd=string&pseudo:string&args:string";
+                json.actualParams = "cmd=" + cmdName + "&pseudo=" + pseudo + "&args=" + args;
+            }
+            else
+            {
+                json.actualParams = "cmd=" + cmdName + "&pseudo=" + pseudo + "&args=" + args;
+                json.status = BLTModule.TwitchService?.TestCommand(cmdName, pseudo, args);
```

Comment:
This will be null of the service isn't initialized (which is the case, e.g. in the main menu). Will json.status null be a problem? I know json can serialize null, but dunno what the implications are for your frontend. I don't think bool can be null. You should write:
`json.status = BLTModule.TwitchService?.TestCommand(cmdName, pseudo, args) ?? false;`
Although I might suggest you return a specific error if the service isn't initialized yet.

Reply:
It's will work, but its not clean. I've revamp the json output to have something clearer.